### ISSUE TITLE
Opam build on 4.02.0 failed

### ISSUE THREAD

**Issue by ontologiae** [the-lambda-church/merlin#355]
$ opam install merlin
      The following actions will be performed:
       - install merlin.pinned
      1 to install | 0 to reinstall | 0 to upgrade | 0 to downgrade | 0 to remove
      
      =-=-= Installing merlin.pinned =-=-=
      merlin.pinned Fetching https://github.com/the-lambda-church/merlin.git#merlin2
      Building merlin.pinned:
        ./configure --prefix /Users/ontologiae/.opam/4.02.0+improved-errors
        rm -rf /Users/ontologiae/.opam/4.02.0+improved-errors/share/ocamlmerlin
        make -j 1
      [ERROR] The compilation of merlin.pinned failed.
      Removing merlin.pinned.
      
      
      ===== ERROR while installing merlin.pinned =====
      # opam-version 1.1.1
      # os           darwin
      # command      make -j 1
      # path         /Users/ontologiae/.opam/4.02.0+improved-errors/build/merlin.pinned
      # compiler     4.02.0+improved-errors
      # exit-code    2
      # env-file     /Users/ontologiae/.opam/4.02.0+improved-errors/build/merlin.pinned/merlin-95546-c61e7b.env
      # stdout-file  /Users/ontologiae/.opam/4.02.0+improved-errors/build/merlin.pinned/merlin-95546-c61e7b.out
      # stderr-file  /Users/ontologiae/.opam/4.02.0+improved-errors/build/merlin.pinned/merlin-95546-c61e7b.err
      ### stdout ###
      # ...[truncated]
      # Will install:
      #   ocamlmerlin binary in: /Users/ontologiae/.opam/4.02.0+improved-errors/bin
      #   vim mode in: /Users/ontologiae/.opam/4.02.0+improved-errors/share/ocamlmerlin/vim
      #
      # To customize directories, use:
      #   ./configure [--prefix <dir>] [--bindir <dir>] [--sharedir <dir>] [--vimdir <vimdir>] [--with-vimbufsync | --without-vimbufsync]
      #
      # For more informations rerun with --help.
      # Now, type 'make' to compile then 'make install'.
      # make -f Makefile.ocamlmakefile WITH_BIN_ANNOT="" WITH_DEBUG="" REALLY_QUIET=1 ocamlmerlin
      ### stderr ###
      # File "src/ocaml/typing/typecore.ml", line 2960, characters 10-32:
      # Error: This variant pattern is expected to have type
      #          (a, b, c, d, e, f) CamlinternalFormatBasics.ignored
      #        The constructor Ignored_scan_next_char does not belong to type CamlinternalFormatBasics.ignored
      # make[2]: *** [src/ocaml/typing/typecore.cmx] Error 2
      # make[1]: *** [native-code] Error 2
      # make: *** [ocamlmerlin] Error 2
      
      'opam install merlin' failed.

**Comment by gasche** [comment]
It works on 4.02.1. Do you have a specific reason to keep using a known-bad version of OCaml?

**Comment by trefis** [comment]
I'm guessing the "+improved-errors" is his reason.
Is that PR available for 4.02.1 ?

It's not the first time we see that error, cf.
- https://github.com/the-lambda-church/merlin/issues/338
- https://github.com/the-lambda-church/merlin/issues/300

But the problem has been fixed since.
I'm guessing ontologiae has a problem because of:
```
  =-=-= Installing merlin.pinned =-=-=
  merlin.pinned Fetching https://github.com/the-lambda-church/merlin.git#merlin2
```

If you unpin, everything should be fine.
